cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/ProgramScheduleHttpHandler.java

```diff
@@ -0,0 +1,813 @@
+import io.cdap.cdap.internal.app.runtime.schedule.ProgramScheduleStatus;
+import io.cdap.cdap.internal.app.runtime.schedule.SchedulerException;
+import io.cdap.cdap.internal.app.runtime.schedule.store.Schedulers;
+import io.cdap.cdap.internal.app.runtime.schedule.trigger.ProgramStatusTrigger;
+import io.cdap.cdap.internal.app.services.ProgramLifecycleService;
+import io.cdap.cdap.internal.app.store.ApplicationMeta;
+import io.cdap.cdap.internal.schedule.constraint.Constraint;
+import io.cdap.cdap.proto.BatchProgram;
+import io.cdap.cdap.proto.BatchProgramSchedule;
+import io.cdap.cdap.proto.ProgramStatus;
+import io.cdap.cdap.proto.ProgramType;
+import io.cdap.cdap.proto.ProtoTrigger;
+import io.cdap.cdap.proto.ScheduleDetail;
+import io.cdap.cdap.proto.ScheduledRuntime;
+import io.cdap.cdap.proto.id.ApplicationId;
+import io.cdap.cdap.proto.id.ApplicationReference;
+import io.cdap.cdap.proto.id.NamespaceId;
+import io.cdap.cdap.proto.id.ProgramId;
+import io.cdap.cdap.proto.id.ProgramReference;
+import io.cdap.cdap.proto.id.ScheduleId;
+import io.cdap.cdap.scheduler.ProgramScheduleService;
+import io.cdap.http.HttpResponder;
+import io.netty.buffer.ByteBufInputStream;
+import io.netty.handler.codec.http.FullHttpRequest;
+import io.netty.handler.codec.http.HttpRequest;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.DefaultValue;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.QueryParam;
+
+@Singleton
+@Path(Constants.Gateway.API_VERSION_3 + "/namespaces/{namespace-id}")
+public class ProgramScheduleHttpHandler extends AbstractAppFabricHttpHandler {
+
+  private static final Type BATCH_PROGRAMS_TYPE = new TypeToken<List<BatchProgram>>() {
+  }.getType();
+  private static final List<Constraint> NO_CONSTRAINTS = Collections.emptyList();
+
+  private final ProgramScheduleService programScheduleService;
+  private final ProgramLifecycleService lifecycleService;
+  private final Store store;
+
+  @Inject
+  public ProgramScheduleHttpHandler(ProgramScheduleService programScheduleService,
+      ProgramLifecycleService lifecycleService,
+      Store store) {
+    this.programScheduleService = programScheduleService;
+    this.lifecycleService = lifecycleService;
+    this.store = store;
+  }
+
+  /**
+   * Returns status of a type specified by the type{flows,workflows,mapreduce,spark,services,schedules}.
```

Comment:
nit: returns status of schedule.

similar comment for all such comments. Please also update them in `ProgramLifecycleHttpHandler`

Reply:
Done